### PR TITLE
feat: Add upgrade support

### PR DESCRIPTION
**Pull request by matsonj**
adding upgrade support in python ext + fixing evidence run dev to work locally + inside codespaces

### CODE REVIEW

File: evidence_ext/extension.py [hunk 1]
```diff
@@ -135,4 +135,29 @@ def dev(self) -> None:
...
+                    self.evidence_home,
+                    "run",
+                    "dev",
+                    "--",
+                    "--host",
+                    "0.0.0.0",
+                ],
+            )
+
+    def upgrade(self, *command_args: ExecArg) -> None:
```
Comment (edgarrmondragon) [inline evidence_ext/extension.py:150]:
```suggestion
    def upgrade(self) -> None:
```

File: evidence_ext/extension.py [hunk 2]
```diff
@@ -135,4 +135,29 @@ def dev(self) -> None:
...
-            self._npm.run_and_log(*["--prefix", self.evidence_home, "run", "dev"])
+            self._npm.run_and_log(
+                *[
+                    "--prefix",
+                    self.evidence_home,
+                    "run",
+                    "dev",
+                    "--",
+                    "--host",
+                    "0.0.0.0",
```
Comment (edgarrmondragon) [inline evidence_ext/extension.py:146]:
```suggestion
                    "0.0.0.0",  # noqa: S104
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]
Thanks @matsonj! A few nits, mostly to make Ruff happy.

**Comment by matsonj** [comment]
@edgarrmondragon Thank you for the help with the CI changes - I haven't used ruff before 😅

**Review by matsonj — COMMENTED** [review]
Looks good now?

**Review by edgarrmondragon — APPROVED** [review]
LGTM. Thanks @matsonj!

**Pull request merged**